Replicator/Replicator.cc

```diff
@@ -106,6 +107,11 @@ namespace litecore { namespace repl {
     }
 
 
+    Replicator::~Replicator() {
+        _weakConnectionDelegateThis->rescind(this);
```

Comment:
More of a comment on WeakHolder but it seems this would be easy to forget.  It would be cleaner if there were a WeakHolder parent of some kind, that can issue weak children, which are auto invalidated when the parent destructs.

Reply:
It's good comment, something like Weakable interface. I would do it next time. My goal this time is to not expose anything beyond the delegate interface. If I put this structure in Weakable parent class, we will have two non-abstract inheritance. I think Scala's Trait is a good language construct for it.